pyiron_gui/project/project_browser.py

```diff
@@ -123,7 +131,363 @@ def _output_conv(self):
+        if groups is None:
+            groups = self.groups + self.nodes if self._node_as_group else self.groups
+        button_list = []
+        for group in groups:
+            button = widgets.Button(description=str(group),
+                                    icon="folder",
+                                    layout=self._item_layout)
+            button.style.button_color = self.color["group"]
+            button.on_click(self._on_click_group)
+            if self._fix_position:
+                button.disabled = True
+            button_list.append(button)
+        return button_list
+
+    @busy_check()
+    def _on_click_node(self, b):
+        self._select_node(b.description)
+        self._update_body_box()
+
+    def _select_node(self, node):
+        if node in self._clicked_nodes:
+            self._clicked_nodes.remove(node)
+            self._data = None
+        else:
+            self._clicked_nodes = [node]
+            try:
+                self._data = self.project[node]
+            except(KeyError, IOError, ValueError):
+                self._data = None
+
+    def _gen_node_buttons(self, nodes=None):
+        if nodes is None:
+            nodes = self.files if self._node_as_group else self.files + self.nodes
+        node_list = []
+        for node in nodes:
+            button = widgets.Button(description=str(node),
+                                    icon="file-o",
+                                    layout=self._item_layout)
+            if node in self._clicked_nodes:
+                button.style.button_color = self.color["file_chosen"]
+            else:
+                button.style.button_color = self.color["file"]
+            button.on_click(self._on_click_node)
+            node_list.append(button)
+        return node_list
+
+    def _update_body_box(self, body_box=None):
+        if body_box is None:
+            body_box = self._body_box
+        if self._fix_position:
+            body_box.children = tuple([WrapingHBox(self._gen_group_buttons()),
+                                       WrapingHBox(self._gen_node_buttons())])
+        else:
+            body_box.children = tuple([widgets.HBox(self._gen_control_buttons()),
+                                       WrapingHBox(self._gen_group_buttons()),
+                                       WrapingHBox(self._gen_node_buttons())])
+
+    def _gen_box_children(self):
+        self._update_body_box()
+        return [self._body_box]
+
+    def refresh(self):
+        """Refresh the project browser."""
+        self._box.children = tuple(self._gen_box_children())
+
+    def gui(self):
+        """Return the VBox containing the browser."""
+        self.refresh()
+        return self._box
+
+    def _ipython_display_(self):
+        """Function used by Ipython to display the object."""
+        display(self.gui())
+
+
+class HasGroupsBrowserWithHistoryPath(HasGroupsBrowser):
+    def __init__(self, project, box=None):
```

Comment:
If I see this correctly the normal `HasGroupsBrowser` already has the infrastructure to keep track of the history, what's the difference to this then?

Reply:
The `HasGroupsBrowser` just keep it's history, while this one constructs an artificial path from the history and displays it in addition.